blockchain/validate.go

```diff
@@ -1152,6 +1152,24 @@ func (b *BlockChain) checkConnectBlock(node *blockNode, block *btcutil.Block, vi
 		runScripts = false
 	}
 
+	// If we're at a checkpoint, then ignore all assumeValid params
+	if checkpoint != nil {
+		// Check if we're at the assumeValidHash. If assumeValidHash isn't nil
+		// don't check signatures. This mimicks the behavior of the reference
+		// client
+		if b.assumeValidHash != nil && node.hash.IsEqual(b.assumeValidHash) {
```

Comment:
Shouldn't this instead be enforced during the headers first sync to ensure that the chain we're about to validate is actually a  _ancestor_  of the specified block hash?

Reply:
Also IIRC, this needs to be coupled w/ a related constant that asserts the minimum about of chain work for the target branch.